### DISCUSSION TITLE
Crash Reporting

### DISCUSSION THREAD

**Discussion by Garrettlop** [Make-Tarkov-Great-Again/MTGA-Backend#92] [General]
I noticed that there wasn't an area to do that. First raid got in just fine ran around looted had an air drop come in almost get kill by a scav with only the gear you can see all of the containers on him were empty was bleeding out ran to the air drop also nothing in the air drop and I then I died then I got an error saying the same item already exists and it crashed I think it had something to do with me buy the item call stash with the dogge texture and I took it in to the raid with me.

**Comment by lavivrus** [comment]
windows powershell shows the server running. launching the client.exe just gives me a black screen. Might have missed something?

**Reply by lavivrus** [reply]
switched over to VS code and got the browser launcher prompts, registered and launched. same thing so far. Is the first start up generally take 5min+?

**Comment by MatthewABrantley** [comment]
I'm able to make my "account" through the webui at :443 but no matter what path I tell it, game won't launch. Been digging through the files trying to add the path somewhere myself but not gotten anywhere.

**Reply by SlejmUr** [reply]
We still waiting for a proper launcher.
I suggest now them to show the argument what needed to send when starting the exe.
Like:
`-config={"Id":"0","Ip":"192.168.1.50"}`
Or smth

**Comment by Capt-GaDam** [comment]
Installed Server and Client and can launch server successfully.  Can create user and login.  
When launching I get the Tarkov splash with spinning wheel that goes on forever without timing out or error.

In consol get the following errors:


[Info   :   BepInEx] Loading [MTGA.Core 1.0.0]
[Error  : Unity Log] ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
Stack trace:
System.Reflection.Assembly.GetTypes () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
MTGA.Core.PatchConstants.get_EftTypes () (at <5e6acf0c9f8845098e93b186759cd407>:0)
MTGA.Core.PatchConstants..cctor () (at <5e6acf0c9f8845098e93b186759cd407>:0)
Rethrow as TypeInitializationException: The type initializer for 'MTGA.Core.PatchConstants' threw an exception.
MTGA.Core.Misc.MongoIDPatch..cctor () (at <5e6acf0c9f8845098e93b186759cd407>:0)
Rethrow as TypeInitializationException: The type initializer for 'MTGA.Core.Misc.MongoIDPatch' threw an exception.
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application:.cctor()
Sirenix.Serialization.UnitySerializationInitializer:Initialize()
Sirenix.Serialization.UnitySerializationInitializer:InitializeRuntime()


[Info   : Unity Log] key:config value:{"BackendUrl":"https://127.0.0.1:443","Version":"live"}
[Error  : Unity Log] ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
Stack trace:
System.Reflection.Assembly.GetExportedTypes () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
GClass2221.Init () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
GClass2221..cctor () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
Rethrow as TypeInitializationException: The type initializer for 'GClass2221' threw an exception.
EFT.CommonClientApplication`1[T].Awake () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
Class296:UnityEngine.ILogHandler.LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)

**Reply by SlejmUr** [reply]
Server branch + Client version?

**Reply by Capt-GaDam** [reply]
Client is 0.13.0.21469
How do I determine "Server branch"?

**Comment by Capt-GaDam** [comment]
![Snag_2d41c40](https://user-images.githubusercontent.com/121689865/210150354-89db6e84-15fd-4da1-ac03-f51c06297d7c.png)

**Reply by Capt-GaDam** [reply]
"name": "MTGA-Server",
  "version": "0.0.1",
  "description": "Make Tarkov Great Again: An `Escape from Tarkov` emulation server built with Fastify",
  "main": "app.js",
  "scripts": {
    "start": "node --trace-warnings app.js",
    "dev": "nodemon --trace-warnings app.js"
  },
  "keywords": [],
  "author": "MTGA-Team",
  "license": "ISC",
  "dependencies": {
    "@fastify/autoload": "^5.6.0",
    "@fastify/compress": "^6.2.0",
    "@fastify/cookie": "^8.3.0",
    "@fastify/formbody": "^7.4.0",
    "@fastify/websocket": "^7.1.1",
    "console-png": "^1.2.1",
    "fast-stringify": "^2.0.0",
    "fastify": "^4.10.2",
    "fastify-plugin": "^4.4.0",
    "mongoid-js": "^1.3.0",
    "selfsigned": "^2.1.1"
  },
  "devDependencies": {
    "nodemon": "^2.0.20",
    "pino-pretty": "^9.1.1",
    "tap": "^16.3.2"
  }
}

**Reply by gixxer85** [reply]
I'm also stuck here... infinite pinwheel and loadingscreen

**Reply by SlejmUr** [reply]
Use the current Tag [21523_KindaPreRel](https://github.com/Make-Tarkov-Great-Again/MTGA-Backend/releases/tag/21523_KindaPreRel) with the latest 0.13.0.1.21523 Client + the core

**Reply by Capt-GaDam** [reply]
Deleted contents of Server folder and extracted 21532_KindaPreRel into it.  Ran: npm install

Launched server and connected with Web.  Created user and logged in successfully.

Launch game and got:
[Info   :   BepInEx] Loading [MTGA.Core 1.0.0]
[Error  : Unity Log] ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
Stack trace:
System.Reflection.Assembly.GetTypes () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
MTGA.Core.PatchConstants.get_EftTypes () (at <5e6acf0c9f8845098e93b186759cd407>:0)
MTGA.Core.PatchConstants..cctor () (at <5e6acf0c9f8845098e93b186759cd407>:0)
Rethrow as TypeInitializationException: The type initializer for 'MTGA.Core.PatchConstants' threw an exception.
MTGA.Core.Misc.MongoIDPatch..cctor () (at <5e6acf0c9f8845098e93b186759cd407>:0)
Rethrow as TypeInitializationException: The type initializer for 'MTGA.Core.Misc.MongoIDPatch' threw an exception.
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application:.cctor()
Sirenix.Serialization.UnitySerializationInitializer:Initialize()
Sirenix.Serialization.UnitySerializationInitializer:InitializeRuntime()



as well as:

[Info   : Unity Log] key:token value:63b098928eba3801700040a3
[Info   : Unity Log] key:config value:{"BackendUrl":"https://127.0.0.1:443","Version":"live"}
[Error  : Unity Log] ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
Stack trace:
System.Reflection.Assembly.GetExportedTypes () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
GClass2221.Init () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
GClass2221..cctor () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
Rethrow as TypeInitializationException: The type initializer for 'GClass2221' threw an exception.
EFT.CommonClientApplication`1[T].Awake () (at <9d6c59c7983f48819c5efd2921ab3be4>:0)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
Class296:UnityEngine.ILogHandler.LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)


See full contents of console:

[console-log.txt](https://github.com/Make-Tarkov-Great-Again/MTGA-Backend/files/10327945/console-log.txt)

**Reply by SlejmUr** [reply]
Bruh.
I hate this.
Currently can't do anything so, @ae-King can you check it?

**Reply by Capt-GaDam** [reply]
During npm install:

M:\MTGA\MTGA-Server>npm install

added 340 packages, and audited 503 packages in 13s

39 packages are looking for funding
  run `npm fund` for details

1 high severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.

M:\MTGA\MTGA-Server>



M:\MTGA\MTGA-Server>npm audit
# npm audit report

json5  <2.2.2
Severity: high
Prototype Pollution in JSON5 via Parse Method - https://github.com/advisories/GHSA-9c47-m6qq-7p4h
fix available via `npm audit fix`
node_modules/tap/node_modules/json5

1 high severity vulnerability

To address all issues, run:
  npm audit fix


M:\MTGA\MTGA-Server>npm audit fix
npm WARN audit fix json5@2.2.1 node_modules/tap/node_modules/json5
npm WARN audit fix json5@2.2.1 is a bundled dependency of
npm WARN audit fix json5@2.2.1 tap@16.3.2 at node_modules/tap
npm WARN audit fix json5@2.2.1 It cannot be fixed automatically.
npm WARN audit fix json5@2.2.1 Check for updates to the tap package.

up to date, audited 503 packages in 1s

39 packages are looking for funding
  run `npm fund` for details

# npm audit report

json5  <2.2.2
Severity: high
Prototype Pollution in JSON5 via Parse Method - https://github.com/advisories/GHSA-9c47-m6qq-7p4h
fix available via `npm audit fix`
node_modules/tap/node_modules/json5

1 high severity vulnerability




Is this of concern?

**Reply by Capt-GaDam** [reply]
Thanks for the assist.  Got it working.  Had to get the Assembly-CSharp.dll and MTGA.Core.dll again from[ [21523] Pre-Release](https://github.com/Make-Tarkov-Great-Again/MTGA-Binaries/releases/tag/21523)

**Comment by Bl1zZ4rD-CH** [comment]
Hey all and thanks for your work :)

I installed it today with the newest client and your wiki's links.

Client: 0.13.0.1.21531

with the dll's postet 3 weeks ago i can start the game but stuck in the loading screen with a spinning wheel.
Then i replaced the dll's with the ones from Capt-GaDam 18 hours ago. with them i get the following message:

    err: {
      "type": "FastifyError",
      "message": "Request body size did not match Content-Length",
      "stack":
          FastifyError: Request body size did not match Content-Length
              at IncomingMessage.onEnd (D:\MTGA-Backend-main\node_modules\fastify\lib\contentTypeParser.js:261:18)
              at IncomingMessage.emit (node:events:513:28)
              at endReadableNT (node:internal/streams/readable:1359:12)
              at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
      "name": "FastifyError",
      "code": "FST_ERR_CTP_INVALID_CONTENT_LENGTH",
      "statusCode": 400
    }

**Reply by Bl1zZ4rD-CH** [reply]
Got it :D followed all the upper Posts steps. now it works, sorry for the post

**Reply by Bl1zZ4rD-CH** [reply]
Okay, seems not to work in game, to many bugs (Cant Load Magazines) Is it normal that you start with lvl 69 and all available? :D

**Reply by Bl1zZ4rD-CH** [reply]
The Game works perfectly now but i dont get any exp or loot, is this normal? ^^

**Comment by MrZavodsky** [comment]
When I launch the game using the website launcher the game loads then stops at a black screen. Eventually I get this Error popup 
![Capture](https://user-images.githubusercontent.com/121979166/210647530-edb9f637-1ecd-422f-a9aa-107c13ccf8e1.PNG)

**Reply by SlejmUr** [reply]
Server branch are develop?

**Reply by MrZavodsky** [reply]
I fixed it thanks for the quick reply. It wasn't but someone on discord helped me

**Comment by foxroxmchw** [comment]
When i click Start Tarkov on browser it says Successful Tarkov will start shortly. Waiting 15min and nothing hapening.

**Reply by T4s3rF4c3** [reply]
This Fork doesn't have this issue.

https://github.com/Rickeetz/MTGA-Backend

**Reply by foxroxmchw** [reply]
Unfortunately same problem with this fork nothing hapening. I tried reinstall everything but still isnt working.

**Reply by Penta19** [reply]
Same for me, says tarkov will start shortly but never starts.

[DEBUG] 2023-01-29 16:29:17 [WEBINTERFACE] Found sessionID cookie: 63d69ec652e3504ae40030d2
[DEBUG] 2023-01-29 16:29:17 [WEBINTERFACE] Launching EscapeFromTarkov.exe...
[DEBUG] 2023-01-29 16:29:17 [WEBINTERFACE] Reading base file: ./assets/templates/webinterface/base.html
[17:29:17.597] INFO (19172): incoming request {"reqId":"req-x","req":"[GET] /webinterface/weblauncher/start"}
[17:29:17.612] INFO (19172): request completed {"reqId":"req-x","res":"200","responseTime":14.651799999177456}
[17:29:17.630] INFO (19172): incoming request {"reqId":"req-y","req":"[GET] /resources/dist/css/bootstrap.min.css"}
[17:29:17.630] INFO (19172): incoming request {"reqId":"req-z","req":"[GET] /resources/navbar-top-fixed.css"}
[17:29:17.630] INFO (19172): incoming request {"reqId":"req-10","req":"[GET] /resources/dist/js/bootstrap.bundle.min.js"}
[17:29:17.631] INFO (19172): incoming request {"reqId":"req-11","req":"[GET] /resources/webinterface.js"}
[17:29:17.634] INFO (19172): request completed {"reqId":"req-z","res":"200","responseTime":3.315600000321865}
[17:29:17.635] INFO (19172): request completed {"reqId":"req-11","res":"200","responseTime":3.9358000010252}
[17:29:17.713] INFO (19172): request completed {"reqId":"req-10","res":"200","responseTime":82.46449999883771}
[17:29:17.890] INFO (19172): request completed {"reqId":"req-y","res":"200","responseTime":259.53099999949336}

**Comment by fw-wlsr** [comment]
Hello Guys

First, thanks for this awesome project 😃 I'm played from time to time with MTGA, but since a few days, I just get these exceptions (over and over) in the client, as soon as i try to start a map or even go into the hideout. I already tried to reinstall the hole client (including compile the client dll's with the newest version) or reset the profile, but no success...

I this a known problem right now or do I something wrong?

Thanks a lot!

![grafik](https://user-images.githubusercontent.com/68393490/215520948-e2243ef8-2577-4936-ad5d-5e021b949eaf.png)

**Reply by fw-wlsr** [reply]
Versions: 
- Backend: develop branch, commit 174a738
- Client: 0.13.0.1.21531

**Comment by yopgaming** [comment]
Hello,

I installed latest versions from source.

Issue : The client starts, then crashes.

Client: 0.13.0.1.22173
MTGA-Binaries: branch: master, commit: aa25d10, date: Mar 10, 2023
MTGA-Backend: branch: develop, commit: 0506ede, date: Mar 10, 2023

```
[Error  :ModulePatch] OfflineSaveProfile:: Method is not found!
[Error  :      MTGA] Plugin: System.InvalidOperationException: OfflineSaveProfile: TargetMethod is null
  at MTGA.Utilities.Core.ModulePatch.Enable () [0x00029] in <5b648f57c119429fb692ef30c37835a8>:0 
  at MTGA.Plugin.Awake () [0x00263] in <5b648f57c119429fb692ef30c37835a8>:0 
```

Best regards

[backend.log](https://github.com/Make-Tarkov-Great-Again/MTGA-Backend/files/10946241/backend.log)
[LogOutput.log](https://github.com/Make-Tarkov-Great-Again/MTGA-Backend/files/10946242/LogOutput.log)

**Reply by SlejmUr** [reply]
It works on the latest, the binaries are not pushed out to be public bot soon I do it.

**Reply by yopgaming** [reply]
Thanks for your replies

**Reply by yopgaming** [reply]
Hello,
I reinstalled everything, following the wiki. It works, I can launch an offline raid.

Client: 0.13.0.1.22173
MTGA-Binaries: [22173] Release
MTGA-Backend: branch: develop, commit: [b472409](https://github.com/Make-Tarkov-Great-Again/MTGA-Backend/commit/b4724097d97e38fbdc14a74f6a5130c8ffdf745e)

I spawned in Interchange, Woods, Factory, Shoreline successfully.

I ran into this issue on Customs : 
Client : "Object reference not set to an instance of an object"
Server
```[23:07:14.559] INFO: request completed {"res":"200","responseTime":0.5996000170707703}
SNIPPET CODE FOR QUICK ADDING INTO THE availableSpawns.json for map Customs
"Loot_key_closet": [],
SNIPPET CODE FOR QUICK ADDING INTO THE availableSpawns.json for map Customs
"NY_ball_red_common": [],
[23:07:34.443] INFO: incoming request {"req":"[POST] /client/location/getLocalloot"}
[23:07:34.453] INFO: incoming request {"req":"[POST] /client/raid/configuration"}
[23:07:34.454] INFO: request completed {"res":"200","responseTime":0.6331999897956848}
[23:07:34.479] INFO: [ACCOUNT SAVE] Account file for profile 641b21f934fdc553ac000001 save skipped!
[23:07:34.482] INFO: [CHARACTER SAVE] Character file for profile 641b21f934fdc553ac000001 save skipped!
[23:07:34.482] INFO: [STORAGE SAVE] Storage file for profile 641b21f934fdc553ac000001 save skipped!
[23:07:34.482] INFO: [DIALOGUE SAVE] Dialogues file for profile 641b21f934fdc553ac000001 save skipped!
[23:07:34.512] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.513] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.513] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.513] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.513] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.513] ERROR: Container: 61a89e5445a2672acf66c877 not found in conrainersSpawnData.json
[23:07:34.516] ERROR: SpawnTable not found for Loot_key_closet with WorldId: Loot_key_closet797554, skipping...
[23:07:34.516] ERROR: SpawnTable not found for NY_ball_red_common with WorldId: NY2022_ball_red_common (7)9172166, skipping...
```

Edit : 
I have the explanation in the directory : assets/database/locations/

Good job, best regards
